tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsContextPropagatorTests.cs

```diff
@@ -38,6 +38,43 @@ public void CanRoundTripPathwayContext()
            .BeCloseTo(FromUnixTimeNanoseconds(context.EdgeStart), oneMs);
     }
 
+    [Fact]
+    public void Inject_WhenLegacyHeadersEnabled_IncludesBothBase64AndBinaryHeaders()
+    {
+        var headers = new TestHeadersCollection();
+        var context = new PathwayContext(
+            new PathwayHash(1234),
+            DateTimeOffset.UtcNow.AddSeconds(-5).ToUnixTimeNanoseconds(),
+            DateTimeOffset.UtcNow.ToUnixTimeNanoseconds());
+
+        DataStreamsContextPropagator.Instance.Inject(context, headers);
+
+        headers.Values.Should().ContainKey(DataStreamsPropagationHeaders.PropagationKeyBase64);
+        headers.Values[DataStreamsPropagationHeaders.PropagationKeyBase64].Should().NotBeNullOrEmpty();
+
+        headers.Values.Should().ContainKey(DataStreamsPropagationHeaders.PropagationKey);
+        headers.Values[DataStreamsPropagationHeaders.PropagationKey].Should().NotBeNullOrEmpty();
```

Comment:
You should also add a test that the legacy header is _not_ injected when disabled.

And you'll probably also want to add tests that assert the headers (both old and new) contain the data we expect and in the correct format. I don't see any tests for that here (the existing round-tripping is not enough to ensure the format is compatible across tracers).